docs/madr/decisions/046-meshservice-policy-matching.md

```diff
@@ -0,0 +1,228 @@
+
+References for `kind: MeshService` have the following structure:
+
+```yaml
+---
+kind: MeshService
+spec:
+  ports:
+  - port: 80
+    name: http
+---
+targetRef:
+  kind: MeshService
+  namespace: backend
+  name: backend # only name or labels
+  labels: # only name or labels
+    kuma.io/zone: east
+  sectionName: http
+```
+
+The `name` of a `MeshService` ref always refers to the name of one, real Kubernetes
+object in the zone of the policy.
+
+Exactly one of either `targetRef.name` or `targetRef.labels` is **required**.
+
+#### `sectionName`
+
+If `sectionName` is set in `targetRef.kind` with `kind: MeshService`,
+it refers to an entry in `ports` by name and only traffic to that port is affected.
+
+In `backendRefs`, `port` is required. There is no `sectionName`, in accordance
+with Gateway API HTTPRoute.
+
+#### `name`
+
+If the policy ref sets `name`, it refers to one `MeshService` in the *local
+zone*.
+
+On a k8s zone:
+* without `namespace`, the policy refers to `MeshServices` in the same
+  namespace as the policy.
+* with `namespace`, only that namespace is searched for matching `MeshServices`
+
+On a universal zone, `namespace` is not valid.
+
+Note that this means it's possible to refer to synced `MeshServices` via their
+transformed `<service>-<hash-suffix>`.
+
+#### `labels`
+
+If `name` is not set, `labels` must be set.
+Via `labels`, groups of `MeshServices` can be matched, for example by:
+
+* `kuma.io/display-name`
+* `kuma.io/zone`
+
+If `kuma.io/zone` is set, a missing `namespace` field means we
+refer to _all_ `MeshServices` with this name.
+If no `kuma.io/zone` is set, the `namespace` is assumed to be the namespace of
+the policy making the reference.
+
+This means we don't infer any similarity between the local namespace
+and the other zone's namespace.
+
+#### Examples
+
+##### `MeshService`
+
+Users can target `MeshServices` in the namespace of the policy
+directly with `name`:
+
+```yaml
+spec:
+ to:
+ - targetRef:
+     kind: MeshService
+     name: backend
+```
+
+or by using `labels` without `name`. The below example searches _only in the
+namespace_ of the policy.
+
+```yaml
+spec:
+ to:
+ - targetRef:
+     kind: MeshService
+     labels:
+       kubernetes.io/service-name: zk
+```
+
+In other zones, we search _all namespaces_:
+
+```yaml
+spec:
+ to:
+ - targetRef:
+     kind: MeshService
+     labels:
+        kuma.io/display-name: backend
+        kuma.io/zone: other-zone
+```
+
+
+##### `backendRef`
+
+The structure of `backendRef` for `MeshService` is largely similar except that
+we *require* a specific port.
+
+```yaml
+kind: MeshHTTPRoute
+spec:
+  targetRef:
+    kind: MeshGateway
+    name: edge-gateway
+  to:
+   - targetRef:
+       kind: Mesh
+     rules:
+       - matches:
+           - path:
+               type: Prefix
+               value: /v1
+         default:
+           backendRefs:
+             - kind: MeshService
+               name: frontend
+               port: http
```

Comment:
Is it always by name? should this be `portName`? 🤔

Reply:
Updated to only allow ints like Gateway API, WDYT?